nixos/modules/tasks/filesystems.nix

```diff
@@ -55,6 +57,19 @@ let
         type = types.listOf nonEmptyStr;
       };
 
+      depends = mkOption {
+        default = [ ];
+        example = [ "/persist" ];
+        type = types.listOf nonEmptyWithoutTrailingSlash;
+        description = ''
+          List of paths that should be mounted before this one. Note that you do
+          not need to add the value of .device or .mountPoint to this list.
```

Comment:
Sounds reasonable

Reply:
See [this comment](https://github.com/NixOS/nixpkgs/pull/86967#discussion_r568693458).

Reply:
Adding "exact" still makes sense.

Also, we could add an assertion that every value in `depends` is a prefix of some `mountPoint`.

Reply:
And why do we even mention `.device`? Only prefixes of mount points make sense in this list.

Reply:
I assume you mean that we could add an assertion every value in `depends` has *some* `mountPoint` as a prefix, and I disagree that we should do this (for the same reason I mentioned [here](https://github.com/NixOS/nixpkgs/pull/86967#discussion_r568771126)).

Plus, if we did this, we would have to exclude `/` (because it's a prefix of *every* `mountPoint`), which would be in some sense "unnatural", which makes me think that we likely don't actually want to do so.

---

Regarding
> And why do we even mention .device? Only prefixes of mount points make sense in this list.

I'm not entirely sure what you mean? `.device` and `.mountPoint` do not need to be added to `.depends` because they're *always* checked, even when not included explicity.

---

I've changed the description slightly, but I agree that the description could be better, so if you have any suggestion for what to write, just say so and I'll update the description.

Reply:
> I'm not entirely sure what you mean? `.device` and `.mountPoint` do not need to be added to `.depends` because they're _always_ checked, even when not included explicity.

I just meant that it wouldn't make sense to add a `device` to `depends` anyways because only `mountPoint`s are checked agains the list.

Reply:
With "You do not need to add the exact value of .device or .mountPoint to this list" I meant that you do not need to add the exact value of .device or .mountPoint of **this filesystem** (not the dependency) to .depends (because `fsBefore` always checks .device and .mountPoint in addition to all elements of .depends).

Reply:
Oh, I see. How about adding "in addition to all filesystems whose mount points are a prefix this filesystem's `device` or `mountPoint`"?

Reply:
See updated suggestion below.